### ISSUE TITLE
Custom  model function examples using async/await or generator/yield (spawn)

### ISSUE THREAD

**Issue by adnan-kamili** [balderdashy/sails#3371]
It would be nice to have custom model examples using generator/yield and async/await on

http://sailsjs.org/documentation/concepts/models-and-orm/models

page to keep the framework sync with best javascript coding practices for asynchronous code..

**Comment by cncolder** [comment]
Generator only make async code look like sync code. But it's not truly 'sync'.
I think if you want sails work with generator friendly. We need rewrite the core logic totally. Just like koajs/koa base on co.

**Comment by adnan-kamili** [comment]
I didn't mean sync code. Their is no point of using sync function in models or even node.js for blocking operations.
It is only about the style of coding (neat style similar to php or c# with try catch). Why can't I write my model functions using es7 async/await in sails. It's javascript!

**Comment by cncolder** [comment]
OK. I got it. In fact generator/yield is not for that. Bcs co by tj. So many people use it for that.
But async/await is ok now. You need active babel or other tools first.

    // ./api/services/TestService.js
    export default {
        async test() {
          let users = await User.find();
          sails.log(users);
        },
    }

**Comment by adnan-kamili** [comment]
Exactly! Actually I wanted similar examples on the sails doc page
http://sailsjs.org/documentation/concepts/models-and-orm/models
Other frameworks have already started using them which makes sails look bit old in this fast growing and changing javascript world.
Sails.js is definitely going to become the rails of node, so it should always look cool :p and latest

**Comment by cncolder** [comment]
Sails base on express3 and some other old version modules to make cool things. I think it's hard to follow new es6 way. You can write es6 code by babel. I use [artificialio/sails-hook-babel ](https://github.com/artificialio/sails-hook-babel) now. But it's not a best choose bcs hook is a bit late chance to active babel in sails. All file in `./config/` can't write in es6. I still find another way to do that. I will try pm2 with `"next_gen_js": true` to see if ok for me.

**Comment by sailsbot** [comment]
Thanks for posting, @adnan-kamili.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by mikermcneil** [comment]
Update: babel is no longer necessary for async/await. And re: express, sails v1 is on express 4. To ask about the latest info or any related questions, see http://sailsjs.com/support
